core/include/detray/propagator/propagator.hpp

```diff
@@ -67,70 +67,53 @@ struct propagator {
-            vector_type<intersection_type> &&candidates = {})
-            : m_param_type(parameter_type::e_free),
-              _stepping(t_in, magnetic_field),
-              _navigation(det, std::move(candidates)) {}
-
-        /// Construct the propagation state with bound parameter
-        DETRAY_HOST_DEVICE state(
-            const bound_track_parameters_type &param, const detector_type &det,
-            vector_type<intersection_type> &&candidates = {})
-            : m_param_type(parameter_type::e_bound),
-              _stepping(param, det),
-              _navigation(det, std::move(candidates)) {}
-
-        /// Construct the propagation state with bound parameter
-        template <typename field_t>
-        DETRAY_HOST_DEVICE state(
-            const bound_track_parameters_type &param,
+            const typename free_track_parameters_type::vector_type &free_vec,
             const field_t &magnetic_field, const detector_type &det,
             vector_type<intersection_type> &&candidates = {})
-            : m_param_type(parameter_type::e_bound),
-              _stepping(param, magnetic_field, det),
+            : _stepping(free_vec, magnetic_field),
               _navigation(det, std::move(candidates)) {}
 
         /// Construct the propagation state from the navigator state view
         DETRAY_HOST_DEVICE state(
-            const free_track_parameters_type &param, const detector_type &det,
-            std::size_t track_index,
+            const typename free_track_parameters_type::vector_type &free_vec,
+            const detector_type &det, std::size_t track_index,
             typename navigator_type::state::view_type nav_view)
-            : m_param_type(parameter_type::e_bound),
-              _stepping(param),
-              _navigation(det, track_index, nav_view) {}
+            : _stepping(free_vec), _navigation(det, track_index, nav_view) {}
 
         /// Construct the propagation state from the navigator state view
         template <typename field_t>
         DETRAY_HOST_DEVICE state(
-            const free_track_parameters_type &param,
+            const typename free_track_parameters_type::vector_type &free_vec,
             const field_t &magnetic_field, const detector_type &det,
             std::size_t track_index,
             typename navigator_type::state::view_type nav_view)
-            : m_param_type(parameter_type::e_bound),
-              _stepping(param, magnetic_field),
+            : _stepping(free_vec, magnetic_field),
               _navigation(det, track_index, nav_view) {}
 
-        DETRAY_HOST_DEVICE
-        parameter_type param_type() const { return m_param_type; }
+        /// Construct the propagation state with bound parameter
+        DETRAY_HOST_DEVICE state(
+            const bound_track_parameters_type &param, const detector_type &det,
+            vector_type<intersection_type> &&candidates = {})
+            : _stepping(param, det), _navigation(det, std::move(candidates)) {}
```

Comment:
Why does the stepper state constructor take the underlying vector in for the free track parameters and the bound parameter type in the other case? Is that consistent?

Reply:
I think the PR implements a good change and I am not meaning why the propagator starts with curvilinear track parameters or what is changed with regard to the covariance transport. What I meant is purely a code design aspect: why are the bound track parameters used with their proper interface of ```bound_track_parameter```, while the free track parameters are bypassed and their internal storage (the free vector) is exposed instead? Why do the constructors of the propagator/stepper take these two parametrizations in two different ways?

So, why do we have one interface as
```
DETRAY_HOST_DEVICE state(
            const bound_track_parameters_type &param, 
 [...]
```
and the other interface as
```
DETRAY_HOST_DEVICE state(
           const typename free_track_parameters_type::vector_type &free_vec,
 [...]
```
instead of 
```
DETRAY_HOST_DEVICE state(
            const free_track_parameters_type &param, 
 [...]
```

Reply:
It is to prevent putting the covariance matrix of free track parameters into the constructor.

Reply:
Ok, I understand now. But we should find a better way of doing that, this feels too much like a hotfix. It seems then, that coupling the covariance to the track parametrisation like this is actually hindering us of writing better interfaces, so we should probably redesign that a bit more thoroughly...